lightning/src/routing/mod.rs

```diff
@@ -9,5 +9,14 @@
 
 //! Structs and impls for receiving messages about the network and storing the topology live here.
 
-pub mod router;
 pub mod network_graph;
+pub mod router;
+pub mod scorer;
+
+/// An interface used to score payment channels for path finding.
+///
+///	Scoring is in terms of fees willing to be paid in order to avoid routing through a channel.
+pub trait Score {
+	/// Returns the fee in msats willing to be paid to avoid routing through the given channel.
+	fn channel_penalty_msat(&self, short_channel_id: u64) -> u64;
```

Comment:
Is there any plan to add something to incentivize a channel that's succeeded? (maybe give them a negative penalty?) Wondering how the new planned `PaymentPathSucceeded` event is gonna affect the Scorer

Reply:
Yeah, had a draft stating the same. ☝️

Reply:
Sounds good, I guess we'll update the docs for that possibility when we get to it